tests/e2e/filter_info/filter_info.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import memgraph
+
+
+def test_label_index_hint(memgraph):
+    memgraph.execute("CREATE (n:Label1:Label2 {prop: 1});")
+    memgraph.execute("CREATE INDEX ON :Label1;")
+
+    # TODO: Fix this test since it should only filter on :Label2 and prop
+    expected_explain = [
+        " * Produce {n}",
+        " * Filter (n :Label1:Label2), {n.prop}",
```

Comment:
What to do with this TODO? I also find it weird that it has to do filtering on both labels

Reply:
yes in the expression with the subclass LabelsTest we have an optimization which has an array of Labels. We have always kept it that way but I encountered this only when I fixed this bug of printing the correct way. So the actual behaviour now is that we will use an indexed scan but we will still test against all the labels which is in my opinion a bug, but I think it's out of scope for now as I'm fixing the printing and not the expression handling.